tests/src/test/scala/system/basic/WskConductorTests.scala

```diff
@@ -0,0 +1,311 @@
+        checkConductorLogsAndAnnotations(nestedActivation, 3) // conductor, step, conductor
+      }
+    }
+
+    // invoke nested conductor with single step, blocking invocation
+    val blockingrun = wsk.action.invoke(
+      conductor,
+      Map(
+        "action" -> conductor.toJson, // invoke nested conductor
+        "params" -> JsObject("action" -> step.toJson), // invoke step (level 1)
+        "n" -> 1.toJson),
+      blocking = true)
+    val activation = wsk.parseJsonString(blockingrun.stdout).convertTo[ActivationResult]
+
+    withClue(s"check failed for blocking conductor activation: $activation") {
+      activation.response.status shouldBe "success"
+      activation.response.result shouldBe Some(JsObject("n" -> 2.toJson))
+      checkConductorLogsAndAnnotations(activation, 3) // conductor, nested conductor, conductor
+    }
+
+    // nested step followed by outer step
+    val secondrun = wsk.action.invoke(
+      conductor,
+      Map(
+        "action" -> conductor.toJson, // invoke nested conductor
+        "state" -> JsObject("action" -> step.toJson), // invoked step on return of nested conductor (level 0)
+        "params" -> JsObject("action" -> step.toJson), // invoke step (level 1)
+        "n" -> 1.toJson))
+    withActivation(wsk.activation, secondrun) { activation =>
+      activation.response.status shouldBe "success"
+      activation.response.result shouldBe Some(JsObject("n" -> 3.toJson))
+      checkConductorLogsAndAnnotations(activation, 5)
+    }
+
+    // two levels of nesting, three steps
+    val thirdrun = wsk.action.invoke(
+      conductor,
+      Map(
+        "action" -> conductor.toJson, // invoke nested conductor
+        "state" -> JsObject("action" -> step.toJson), // invoke step on return (level 0)
+        "params" -> JsObject(
+          "action" -> conductor.toJson, // invoked nested nested conductor
+          "state" -> JsObject("action" -> step.toJson), // invoke step on return (level 1)
+          "params" -> JsObject("action" -> step.toJson)), // invoke step (level 2)
+        "n" -> 1.toJson))
+    withActivation(wsk.activation, thirdrun) { activation =>
+      activation.response.status shouldBe "success"
+      activation.response.result shouldBe Some(JsObject("n" -> 4.toJson))
+      checkConductorLogsAndAnnotations(activation, 5)
+    }
+  }
+
+  /**
+   * checks logs for the activation of a conductor action (length/size and ids)
+   * checks that the cause field for nested invocations is set properly
+   * checks duration
+   * checks memory
+   */
+  private def checkConductorLogsAndAnnotations(activation: ActivationResult, size: Int) = {
+    activation.logs shouldBe defined
+    // check that the logs are what they are supposed to be (activation ids)
+    // check that the cause field is properly set for these activations
+    activation.logs.get.size shouldBe (size) // the number of activations in this sequence
```

Comment:
I think you can rewrite this as `activation.logs.get should have length size`

Reply:
OK. I will try.